lib/modules/manager/helmfile/extract.ts

```diff
@@ -42,21 +42,31 @@ export async function extractPackageFile(
   } catch (err) {
     logger.debug(
       { err, packageFile },
-      'Failed to parse helmfile helmfile.yaml',
+      `Failed to parse helmfile ${packageFile}`,
     );
     return null;
   }
-  for (const doc of docs) {
-    if (!(doc && is.array(doc.releases))) {
+  for (const [docIndex, doc] of docs.entries()) {
+    if (!doc) {
       continue;
     }
 
+    // Always check for repositories in the current document and override the existing ones if any (as YAML does)
     if (doc.repositories) {
+      registryAliases = {};
```

Comment:
Does Helmfile really enforces here an order to the documents?

Reply:
Hi @secustor No, Helmfile does not enforce an order. I gave a long explanation in the PR description, it has to do with the fact that YAML would override earlier occurrences of the repositories list.

Another way to handle this would be to fully merge all documents together before extracting repositories and releases.

Reply:
What does `helmfile` do? Does it merge? Does this work or is this invalid?

```yaml
repositories:
 - name: prometheus-community
   url: https://prometheus-community.github.io/helm-charts

---
repositories:
 - name: bitnami
   url: https://charts.bitnami.com/bitnami

releases:
- name: prom-norbac-ubuntu
  namespace: prometheus
  chart: prometheus-community/prometheus
  ## Add a version to the example
  version: 25.7.0
  set:
  - name: rbac.create
    value: false
```

Reply:
Quoting my PR description:

> so e.g. a repositories list defined in the first document would completely be overridden by a repositories list in a later document

Yes, this works. Helmfile merges, but it does not concatenate lists.

Reply:
ok, so my sample won't work because the second `repositories` overwrites the first and then helmfile can't find the `prometheus-community/prometheus` chart?